packages/evolution-backend/src/api/helpers/downloadCsvFile.ts

```diff
@@ -52,16 +60,24 @@ export const downloadCsvFile = async ({
                     reject(error);
                 })
                 .on('data', (row) => {
-                    // Prepare the CSV row with default values
-                    const csvRow: { [key: string]: string | number | boolean | null } = {
-                        interview_id: row.interview_id,
-                        user_email: row.user_email,
-                        user_username: row.user_username
-                    };
+                    const csvRow: { [key: string]: string | number | boolean | null } = {};
+
+                    // Add if not restricted
+                    if (!excludeFields?.includes('interview_id')) {
+                        csvRow.interview_id = row.interview_id;
+                    }
+                    if (!excludeFields?.includes('user_email')) {
+                        csvRow.user_email = row.user_email;
+                    }
+                    if (!excludeFields?.includes('user_username')) {
+                        csvRow.user_username = row.user_username;
+                    }
 
-                    requestedResponses.forEach((response) => {
-                        // We use lodash's get function to safely access nested properties
-                        csvRow[response] = get(row.responses, response, null);
+                    // Add requested responses except restricted responses
```

Comment:
So this comment implies that a given field can be both requested AND excluded? Sounds counter-productive...

Reply:
Okay, j'ai corrigé.